number/backorder.go

```diff
@@ -0,0 +1,118 @@
+package number
+
+import (
+	"fmt"
+	messagebird "github.com/messagebird/go-rest-api/v9"
+	"net/http"
+)
+
+type BackOrderID string
+
+type Backorder struct {
+	ID          string
+	ProductID   int
+	Country     string
+	Prefix      string
+	Status      string
+	ReasonCodes []string
+}
+
+type BackorderDocument struct {
+	ID          int
+	Name        string
+	Description string
+	Status      string
+}
+
+type BackorderDocuments struct {
+	Limit, Count int
+	Items        []*BackorderDocument
+}
+
+type EndUserDetail struct {
+	ID    string
+	Label string
+}
+
+type EndUserDetails struct {
+	Items []*EndUserDetail
+}
+
+type PlaceBackorderRequest struct {
+	ProductID int    `json:"productID"`
+	Prefix    string `json:"prefix"`
+	Quantity  int    `json:"quantity"`
+}
+
+type CreateBackorderDocumentRequest struct {
+	ID       int    `json:"id"`
+	Name     string `json:"name"`
+	MimeType string `json:"mimeType"`
+	Content  string `json:"content"`
+}
+
+type CreateBackorderEndUserDetailRequest struct {
```

Comment:
This struct doesn't have json tags because it has same request structure.
[From reference](https://developers.messagebird.com/api/numbers/#cancel-a-purchased-phone-number):
```
curl -X POST "https://numbers.messagebird.com/v1/backorders/48f6057c21de42d4bbf73fb86caaf361/end-user-details" \
  -H "Authorization: AccessKey test_gshuPaZoeEG6ovbc8M79w0QyM" \
  -H "Content-Type: application/json" \
  -d '{
    "CompanyName": "Messagebird",
    "Street": "Burgwal",
    "StreetNumber": "35",
    "ZipCode": "2011",
    "City": "Amsterdam",
    "Country": "Netherlands"
    }'
```